src/TestProblem/Hydro/ClusterMerger/Flu_ResetByUser_ClusterMerger.cpp

```diff
@@ -0,0 +1,1132 @@
+// problem-specific global variables
+// =======================================================================================
+// (1) read from Input__TestProb
+extern int        Merger_Coll_NumHalos;
+extern double    *CM_BH_Mass;
+extern double    *Jet_HalfHeight;
+extern double    *Jet_Radius;
+
+extern bool       AGN_feedback;
+extern int        Accretion_Mode, JetDirection_case;
+extern double     eta, eps_f, eps_m, R_acc, R_dep;
+extern bool       fixBH, AdjustBHPos, AdjustBHVel;
+extern double     AdjustPeriod;
+// ---------------------------------------------------------------------------------------
+// (2) read from files
+extern int        JetDirection_NBin;                       // number of bins of the jet direction table
+extern double    *CM_Jet_Time_table;                       // the time  table of jet direction
+extern double   **CM_Jet_Theta_table;                      // the theta table of jet direction for 3 clusters
+extern double   **CM_Jet_Phi_table;                        // the phi   table of jet direction for 3 clusters
+// ---------------------------------------------------------------------------------------
+// (3) variables to record
+extern double   (*CM_ClusterCen)[3];
+extern double   (*CM_BH_Pos)[3];                           // BH position (for updating CM_ClusterCen)
+extern double   (*CM_BH_Vel)[3];                           // BH velocity
+extern double    *CM_BH_Mdot_tot;                          // the total accretion rate
+extern double    *CM_BH_Mdot_hot;                          // the hot   accretion rate
+extern double    *CM_BH_Mdot_cold;                         // the cold  accretion rate
+extern double    *CM_Jet_Mdot;                             // the feedback injection rate
+extern double    *CM_Jet_Pdot;
+extern double    *CM_Jet_Edot;
+extern double   (*CM_Jet_Vec)[3];                          // jet direction
+extern double   (*CM_RAcc_GasVel)[3];                      // gas velocity
+extern double    *CM_RAcc_SoundSpeed;
+extern double    *CM_RAcc_GasDens;
+extern double    *CM_RAcc_RelativeVel;                     // the relative velocity between BH and gas
+extern double    *CM_RAcc_ColdGasMass;
+extern double    *CM_RAcc_GasMass;
+extern double    *CM_RAcc_ParMass;
+extern int       *CM_Cluster_NPar_close;
+extern double    *CM_Bondi_SinkMass, *CM_Bondi_SinkMomX, *CM_Bondi_SinkMomY, *CM_Bondi_SinkMomZ;
+extern double    *CM_Bondi_SinkMomXAbs, *CM_Bondi_SinkMomYAbs, *CM_Bondi_SinkMomZAbs;
+extern double    *CM_Bondi_SinkE, *CM_Bondi_SinkEk, *CM_Bondi_SinkEt;
+extern int       *CM_Bondi_SinkNCell;
+// ---------------------------------------------------------------------------------------
+// (4) other variables
+extern double    *E_inj_exp;                               // the expected amount of injected energy
+extern double    *M_inj_exp;                               // the expected amount of injected gas mass
+extern double   (*ang_mom_sum)[3];
```

Comment:
The floating-point types of `ang_mom_sum`, `ang_mom` , and `MPI_GAMER_REAL` are inconsistent.
- Specifically, `ang_mom_sum` uses `double` while `ang_mom` uses `real`. Also, note that `MPI_GAMER_REAL` maps to `real`.
- I suggest always using `double` for `ang_mom` and all other variables storing cumulative results (i.e., any variables passed to `MPI_Allreduce(MPI_SUM)` ), since they are more sensitive to rounding errors.
- Accordingly, change `MPI_GAMER_REAL` to `MPI_DOUBLE`.

Reply:
Merged